src/sentry/integrations/bitbucket_server/webhook.py

```diff
@@ -69,9 +70,17 @@ def __call__(self, organization, integration_id, event):
 
         for change in event["changes"]:
             from_hash = None if change.get("fromHash") == "0" * 40 else change.get("fromHash")
-            for commit in client.get_commits(
-                project_name, repo_name, from_hash, change.get("toHash")
-            ):
+            try:
+                commits = client.get_commits(
+                    project_name, repo_name, from_hash, change.get("toHash")
+                )
+            except ApiUnauthorized:
+                return HttpResponse(status=400)
```

Comment:
Do we want to respond with a 409 to prevent redelivery of the webhook?

Reply:
Any 4xx status code doesn't get retry the webhook outbox message.